### PR TITLE
v2.0.0

### LINKED ISSUE #29: Add `target_variable` and `target_outcome` to task ids as  potential task ids
This reflects the fact we include target as a potential `task_id` and `target_variable` and `target_outcome` are an option that has been discussed.

One question I had is that while using a single `target` task id, optional and required arrays of multiple unique targets are explicit, what happens when you have multiple targets but spread across  `target_variable` and `target_outcome`. How do you ensure they specify unique and correct combinations? Does the order they are specified in the array become important? 

e.g. if two valid targets where `inc hosp` & `inc case`, how would they be specified?

```
"target_variable": {
  "required": null,
  "optional": ["hosp", "case"]
},
"target_outcome": {
  "required": null,
  "optional": ["inc", "inc"]
},
``` 

@elray1  your thoughts would be appreciated!

### LINKED ISSUE #43: Use`uniqueItems: true` to ensure arrays contain unique values

### LINKED ISSUE #44: Ensure `$schema` properties start with `http` not `https`
URLs which start with `https` are not recognised by `ajv` causing issues when using other validators like `ajv-cli`. So all URLs in `$schema` properties specified the json schema version should start with `http`.

See issues brought up in #42

### LINKED ISSUE #53: Add `forecast_date` / `target_end_date` properties to schema?
In our schema we currently have a `origin_date`:

https://github.com/Infectious-Disease-Modeling-Hubs/schemas/blob/c1d8fc0e993ed07c2d783becbe91ea10e31c7e85/v1.0.0/tasks-schema.json#L51C42-L60

and a `target_date` property:

https://github.com/Infectious-Disease-Modeling-Hubs/schemas/blob/c1d8fc0e993ed07c2d783becbe91ea10e31c7e85/v1.0.0/tasks-schema.json#L268-L306

Have we settled on whether `forecast_date` (or other) / `target_end_date` represent standard concepts and should therefore have schema properties for validating them when included in `tasks.json` files?

### LINKED ISSUE #54: Add schema for `task_id` additional properties
Set a schema for additional task id properties. See example here: https://json-schema.org/understanding-json-schema/reference/object.html#additional-properties

### LINKED ISSUE #55: rename `type_id` to `output_type_id`
e.g. see [this line](https://github.com/Infectious-Disease-Modeling-Hubs/schemas/blob/c1d8fc0e993ed07c2d783becbe91ea10e31c7e85/v1.0.0/tasks-schema.json#L403)

[Elsewhere](https://github.com/Infectious-Disease-Modeling-Hubs/hubUtils/issues/71), we have decided to name this `output_type_id` rather than `type_id`, but it seems like the schemas here have not been updated yet. Is this intentional, or just an oversight?

### PR DESCRIPTION
**Pull request by annakrystalli**
* Major breaking change: Output type property `type_id` renamed to `output_type_id` for consistency in with changes in model output data (Resolves #55).
* Added `uniqueItem` checks to all `optional` and `required` arrays (Resolves #43).
* Added a schema to `task_ids` `additionalProperty` property instead of just setting it to `true`. This schema ensures that any custom task IDs added by hub administrators at least checked for being objects and having `optional` and `required` properties that are either arrays or null (Resolves #54).
* Added standard task IDs `forecast_date` (equivalent to `origin_date`), `target_end_date` (equivalent to `target_date`) (Resolves #53).
* Resolves #44 
* Resolves #29 
* Adds some better examples to `target` task ID.

### CODE REVIEW

**Comment by annakrystalli** [comment]
Let know if you would also like me to include #29 in this PR.

Also, if there are any data types associated with any of the current standard task IDs that don't look right / need changes also let me know.

And any other comments welcome! Good opportunity to correct anything that's been bugging us!

**Comment by elray1** [comment]
would it be worth tackling #44 as part of this?

File: v2.0.0/tasks-schema.json [hunk 1]
```diff
@@ -0,0 +1,1338 @@
...
+                                                    }
+                                                }
+                                            },
+                                            "required": [
+                                                "required",
+                                                "optional"
+                                            ]
+                                        },
+                                        "forecast_date": {
+                                            "description": "An object containing arrays of required and optional unique forecast dates. Forecast date usually defines the starting point that can be used for calculating a target_date via the formula target_date = forecast_date + horizon x time_units_per_horizon (e.g., with weekly data, target_date is calculated as forecast_date + horizon x 7 days)",
```
Comment (elray1) [inline v2.0.0/tasks-schema.json:94]: Historically, a forecast date has been the date that a forecast was run or submitted to a hub, which is different than a kind of reference date or origin date for calculating target dates. I'm editing this description here to reflect the way this has been used in the past.
```suggestion
                                            "description": "An object containing arrays of required and optional unique forecast dates. Forecast date usually defines the date that a model is run to produce a forecast.",
```

Comment (nickreich) [reply v2.0.0/tasks-schema.json:94]: @elray1 's suggestion looks good to me.

**Comment by annakrystalli** [comment]
OK! I've incorporated all issues I could. Note I had to backtrack on #44 (details in issue) but feel we should close it.

If you are happy, feel free to leave an approving review and I will go ahead and merge.

**Comment by annakrystalli** [comment]
Morning @nickreich & @elray1 ,

Any chance you good give the OK on this PR? I need to release this so to update all test materials and test snapshots in `hubUtils` with respect to the breaking `output_type_id` change.

**Review by elray1 — APPROVED** [review]

**Pull request merged**